app/src/main/java/de/westnordost/streetcomplete/data/osmtracks/TracksApiImpl.kt

```diff
@@ -60,3 +59,9 @@ private inline fun <T> wrapExceptions(block: () -> T): T =
         // request timeout is a temporary connection error
         throw if (e.errorCode == 408) ConnectionException(e.message, e) else e
     }
+
+private fun LocalDateTime.toTrackFilename(): String {
+    fun Int.f(len: Int): String = this.toString().padStart(len, '0')
+    return ("${this.year.f(4)}_${this.monthNumber.f(2)}_${this.dayOfMonth.f(2)}"
+        + "T${this.hour.f(2)}_${this.minute.f(2)}_${this.second.f(2)}.${this.nanosecond.f(6)}Z.gpx")
```

Comment:
No need for `this`
```suggestion
    fun Int.f(len: Int): String = toString().padStart(len, '0')
    return ("${year.f(4)}_${monthNumber.f(2)}_${dayOfMonth.f(2)}"
        + "T${hour.f(2)}_${minute.f(2)}_${second.f(2)}.${nanosecond.f(6)}Z.gpx")
```

Reply:
https://github.com/streetcomplete/StreetComplete/pull/4471/commits/0466eace8700043627df4c27c1c5a0f76bc4b84e